sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/TableWithoutHeaderCheck.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SonarHTML :: SonarQube Plugin
+ * Copyright (c) 2010-2019 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.sonar;
+
+import java.util.Deque;
+import java.util.LinkedList;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S5256")
+public class TableWithoutHeaderCheck extends AbstractPageCheck {
+
+  private static class TableEntry {
+
+    int line;
+    boolean ignored;
+    boolean hasHeader;
+
+    public TableEntry(int line, boolean ignored) {
+      this.line = line;
+      this.ignored = ignored;
+      this.hasHeader = false;
+    }
+  }
+
+  private Deque<TableEntry> tables = new LinkedList<>();
```

Comment:
I feel this solution using `Deque` and keeping states of seen `table` elements is a bit overly complicated.
You could get the same result by checking `if (isTable(node) && node.getChildren().stream().noneMatch(TableWithoutHeaderCheck::isTableHeader))` inside the `startElement` method, and checking recusively for `aria-hidden` attribute of parent nodes

Reply:
I relied on `Deque` because I wrongly considered `<table>` having headers as long as it has a `<th>` somewhere. However, it seems that headers can only be located within a `<tr>`. I will change that then.

Regarding the `aria-hidden` thing, I like that idea; it may be a bit expensive but definitely more elegant.